rage/vector.hpp

```diff
@@ -27,22 +27,133 @@ namespace rage
 		T data[3];
 		struct { T x, y, z; };
 
-		vector3(T x, T y, T z) :
-			x(x),
+		vector3(T x, T y, T z) noexcept : x(x),
 			y(y),
 			z(z)
 		{
 		}
 
-		vector3() :
-			x(),
+		vector3() noexcept : x(),
 			y(),
 			z()
 		{
 		}
+
+		vector3 operator+(const vector3 &rhs) const
+		{
+			return vector3(x + rhs.x, y + rhs.y, z + rhs.z);
+		}
+
+		vector3 operator-(const vector3 &rhs) const
+		{
+			return vector3(x - rhs.x, y - rhs.y, z - rhs.z);
+		}
+
+		vector3 operator*(const vector3 &rhs) const
+		{
+			return vector3(x * rhs.x, y * rhs.y, z * rhs.z);
+		}
+
+		vector3 operator/(const vector3 &rhs) const
+		{
+			return vector3(x / rhs.x, y / rhs.y, z / rhs.z);
+		}
+
+		vector3 operator+(const float &rhs) const
+		{
+			return vector3(x + rhs, y + rhs, z + rhs);
+		}
+
+		vector3 operator-(const float &rhs) const
+		{
+			return vector3(x - rhs, y - rhs, z - rhs);
+		}
+
+		vector3 operator*(const float &rhs) const
+		{
+			return vector3(x * rhs, y * rhs, z * rhs);
+		}
+
+		vector3 operator/(const float &rhs) const
+		{
+			return vector3(x / rhs, y / rhs, z / rhs);
+		}
+
+		bool operator==(const vector3 &rhs) const
+		{
+			return x == rhs.x && y == rhs.y && z == rhs.z;
+		}
+
+		vector3 &operator+=(const vector3 &rhs)
+		{
+			return *this = *this + rhs;
+		}
+
+		vector3 &operator-=(const vector3 &rhs)
+		{
+			return *this = *this - rhs;
+		}
+
+		vector3 &operator*=(const vector3 &rhs)
+		{
+			return *this = *this * rhs;
+		}
+
+		vector3 &operator/=(const vector3 &rhs)
+		{
+			return *this = *this / rhs;
+		}
+
+		vector3 &operator+=(const float &rhs)
+		{
+			return *this = *this + rhs;
+		}
+
+		vector3 &operator-=(const float &rhs)
+		{
+			return *this = *this - rhs;
+		}
+
+		vector3 &operator*=(const float &rhs)
+		{
+			return *this = *this * rhs;
+		}
+
+		vector3 &operator/=(const float &rhs)
+		{
+			return *this = *this / rhs;
+		}
+
+		float length() const
```

Comment:
This might be pedantic, but this should really be called magnitude, or maybe just add a function called magnitude that calls this. I've heard terms used interchangeably such as Size/Length/Magnitude/Absolute Value/Dot Product/etc.